tests/test_server_basic.py

```diff
@@ -471,14 +471,101 @@ def test_query_restricted_instrument(dispatcher_live_fixture):
     assert jdata["error_message"] == "Unfortunately, your priviledges are not sufficient to make the request for this instrument.\n"
 
 
+@pytest.mark.fast
+@pytest.mark.parametrize("allow_redirect", [True, False])
+def test_instrument_list_redirection(dispatcher_live_fixture_with_local_products_url, allow_redirect):
+    server = dispatcher_live_fixture_with_local_products_url
+
+    logger.info("constructed server: %s", server)
+
+    c = requests.get(os.path.join(server, "api/instr-list"), allow_redirects=allow_redirect)
+
+    if not allow_redirect:
+        assert c.status_code == 302
+        redirection_url = c.headers["Location"]
+        assert "instr-list" in redirection_url
```

Comment:
dont you want to check that it uses the config url and not the bind host?

Reply:
This test was indeed not really exhaustive, I am adding more

Reply:
now it's tested